sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/TwinAsyncTests.java

```diff
@@ -0,0 +1,103 @@
+
+public class TwinAsyncTests extends TwinTestBase
+{
+    private final ClientLogger logger = new ClientLogger(TwinAsyncTests.class);
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.digitaltwins.core.TestHelper#getTestParameters")
+    @Override
+    public void DigitalTwins_Lifecycle(HttpClient httpClient, DigitalTwinsServiceVersion serviceVersion)
+    {
+        DigitalTwinsAsyncClient asyncClient = getAsyncClient(httpClient, serviceVersion);
+
+        String roomTwinId = UniqueIdHelper.getUniqueDigitalTwinId(TestAssetDefaults.ROOM_TWIN_ID_PREFIX, asyncClient, randomIntegerStringGenerator);
+        String floorModelId = UniqueIdHelper.getUniqueModelId(TestAssetDefaults.FLOOR_MODEL_ID, asyncClient, randomIntegerStringGenerator);
+        String roomModelId = UniqueIdHelper.getUniqueModelId(TestAssetDefaults.ROOM_MODEL_ID, asyncClient, randomIntegerStringGenerator);
+
+        String roomTwin = TestAssetsHelper.getRoomTwinPayload(roomModelId);
+        String roomModel = TestAssetsHelper.getRoomModelPayload(roomModelId, floorModelId);
+        List<String> modelsList = new ArrayList<>(Arrays.asList(roomModel));
+
+        try {
+            // Create models to test the Twin lifecycle.
+            StepVerifier
+                .create(asyncClient.createModels(modelsList))
+                .assertNext(createResponseList -> logger.info("Created {} models successfully", createResponseList.size()))
+                .verifyComplete();
+
+            // Create a Twin
+            StepVerifier.create(asyncClient.createDigitalTwin(roomTwinId, roomTwin, BasicDigitalTwin.class))
+                .assertNext(createdTwin -> {
+                    assertEquals(createdTwin.getId(), roomTwinId);
+                    logger.info("Created {} twin successfully", createdTwin.getId());
+                })
+                .verifyComplete();
+
+            // Get a Twin
+            StepVerifier.create(asyncClient.getDigitalTwinWithResponse(roomTwinId))
+                .assertNext(getResponse -> {
+                    assertEquals(getResponse.getStatusCode(), HttpURLConnection.HTTP_OK);
+                    logger.info("Got Twin successfully");
+
+                })
+                .verifyComplete();
+
+            // Update Twin
+            StepVerifier.create(asyncClient.updateDigitalTwinWithResponse(roomTwinId, TestAssetsHelper.getRoomTwinUpdatePayload(), new UpdateDigitalTwinRequestOptions()))
+                .assertNext(updateResponse -> {
+                    assertEquals(updateResponse.getStatusCode(), HttpURLConnection.HTTP_NO_CONTENT);
+                    logger.info("Updated the twin successfully");
+                })
+                .verifyComplete();
+        }
+        // clean up
+        finally {
+            try
+            {
+                if (roomTwinId != null)
+                {
+                    asyncClient.deleteDigitalTwin(roomTwinId);
```

Comment:
this call needs to be blocked, else the process will exit before the async call completes.

Reply:
I had the same comment. I will fix this in another PR.